java-test-samples/fargate-dynamodb-sync/README.md

```diff
@@ -0,0 +1,82 @@
+# Synchronous Approach- Springboot with DynamoDB Application
+Synchronous sample SpringBoot application written in Java which performs customer CRUD operations. AWS serverless services like Amazon API Gateway, ALB, ECR, Fargate, DynamoDB & SQS being used to host and test the application flow.
+Java application is containerized & docker image pushed to ECR. Application is deployed into Fargate & endpoints are available through Application load balancer. API endpoints are set in API Gateway. CRUD operations are invoked via API Gateway which invokes API endpoints of application. Customer details sent to API requests are stored into DynamoDB tables as well as accessed as per the CRUD operations. Finally, post Deployment event placed into SQS.
+
+![customer_crud_app.png](doc%2Fcustomer_crud_app.png)
+
+### Application
+Java17 maven application with Springboot 2.2.6 and AWS Java SDK 1.11.614 dependencies.
+
+### Pre-requisites
+1) SQS Queue
+2) DynamoDB tables
+
+### Note
+Once the application starts, we will trigger a message {"application_status":"ready"} will be dropped to the SQS
+
+_________________________________________________________________________________________________________
+## Integration Testing:
+
+Serverless Application Development Approach:
+The development of each application is based on AWS serverless Architecture. All Microservices are running in ECS Cluster as Fargate Containers. As part of micro service deployment template, required CPU core, memory, number of containers and auto scaling configuration are defined and same is provisioned during deployment. Each micro service container is associated with an ingress container which act as application load balancer.
+
+![AWS_Lambda.png](doc%2FAWS_Lambda.png)
+
+1. Once the deployment is done successfully in ECS, there will be message sent to SNS topic. 
+2. As part of Lambda project, SQS will be created and subscribes to SNS topic. 
+3. Lambda has all required access policies to run the automation test scripts defined to test the serverless application. 
+4. Once the execution is done, the html report will be generated and stored in S3 bucket for further validation. 
+
+### Steps to follow for Lambda set-up
+1.	Set-up SNS project
+2.	Set-up Lambda Project
+3.	Deploy the lambda & test scripts code every time new tests were added.
+4.	Lambda deployment will deploy the latest automation tests in Lambda
+5.	In continuous deployment tool set notification message to SNS post deployment
+
+### Manual trigger in Lambda: 
+    Once the latest tests are deployed, navigate to Lambda function in AWS console, and click the Run button.
+### Auto trigger in Lambda: 
+Since the SQS subscribed to SNS topic, whenever the deployment happens, the Lambda will trigger the automation tests to test the microservices. 
+
+### Steps to follow in PyTest-Python:
```

Comment:
Ideally, I have not seen a Java application using Python as a e2e test suite framework. Is there a rationale of choosing Python instead of other test frameworks like Selenium, Cucumber, Geb, etc?

Reply:
No rational behind in choosing Python. Same can be done by Java, RestAssured, Cucumber combination as well. Since there is no direct dependency on language of an application for e2e testing, We considered application development & testing as 2 different units of work, even in the enterprises we have used golang as well to perform the tests

Reply:
No rational behind in choosing Python. Same can be done by Java, RestAssured, Cucumber combination as well. Since there is no direct dependency on language of an application for e2e testing, we considered application development & testing as 2 different units of work.